### DISCUSSION TITLE
Could not shut down Debian after install k3s

### DISCUSSION THREAD

**Discussion by alvin-7** [k3s-io/k3s#8494] [Q&A] [answered]
**Environmental Info:**
K3s Version: 

k3s version v1.28.2+k3s1 (6330a5b4)
go version go1.20.8

Node(s) CPU architecture, OS, and Version: 
Linux debian 6.1.0-12-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.52-1 (2023-09-07) x86_64 GNU/Linux

Cluster Configuration:
1server

**Describe the bug:**
I installed a Debian container (not LXC) on pve, and after installing k3s on Debian, Debian could not shut down anymore. The effect of shutdown was that memory usage was around 50%, CPU usage was 0, but it could not successfully shut down.

If k3s was not installed, Debian could shut down normally.

**Steps To Reproduce:**

- Installed K3s: 
curl -sfL https://rancher-mirror.rancher.cn/k3s/k3s-install.sh |INSTALL_K3S_CHANNEL=latest INSTALL_K3S_MIRROR=cn INSTALL_K3S_EXEC="--disable=traefik" sh -
- shutdown Debian
- show this issue

**Expected behavior:**
Success to shutdown debian.

**Actual behavior:**
 Debian could not shut down anymore. The effect of shutdown was that memory usage was around 50%, CPU usage was 0, but it could not successfully shut down

**Additional context / logs:**

**Comment by brandond** [comment]
Can you show what specifically is preventing shutdown? K3s will respond to termination signals, same as any other process.

Some Linux distros will pause for a while trying to unmount all the pod volumes, but there are existing discussions about how to address that by killing pods before shutting down.

**Reply by alvin-7** [reply]
I don't know what prevent shutdown
When I try to stop the container, I see PVE told me timeout to stop container.

I am facing the same issue when trying to stop the container using PVE's LXC feature. If I try to shut down the lxc normally, it fails with a timeout error. The only way to successfully stop it is using kill -9. At first I thought it was an issue with LXC, but later when I switched to PVE's container system, I still couldn't stop the container properly.
![image](https://github.com/k3s-io/k3s/assets/38536867/059bcc5e-7108-4262-81ac-1d24eac38833)

**Reply by alvin-7** [reply]
I have also tried using version v1.27.6+k3s1, and encountered the same issue. Even after upgrading to v1.28.2+k3s1 now, I still cannot shut down the container properly.

**Reply by brandond** [reply]
I would probably just run a "shutdown" command from within the host, and watch the console to see where it gets stuck?

**Reply by alvin-7** [reply, marked as answer]
This issue was resolved because my deployment was using an NFS mount, which caused the lxc shutdown to fail. I was using NFS mounts from the PVE system itself and map into the Debian LXC container. When I changed the deployment to use hostPath mounts instead, the issue went away.
